presto-main/src/main/java/com/facebook/presto/type/DecimalParametricType.java

```diff
@@ -37,26 +36,22 @@ public String getName()
     }
 
     @Override
+    @NotNull
     public Type createType(List<TypeParameter> parameters)
     {
+        checkArgument(parameters.size() <= 2, "Expected no more than 2 literal parameters for DECIMAL");
+        Type result = null;
+
         if (parameters.size() == 0) {
-            // we need support for that for support of unparametrized TypeSignatures which are
-            // used for defining DECIMAL related functions. See e.g. DecimalOperators class.
-            return ShortDecimalType.createUnparametrizedDecimal();
+            result = ShortDecimalType.createDecimalType();
+        }
+        else if (parameters.size() == 1) {
+            result = ShortDecimalType.createDecimalType(parameters.get(0).getLongLiteral().intValue());
+        }
+        else if (parameters.size() == 2) {
+            result = ShortDecimalType.createDecimalType(parameters.get(0).getLongLiteral().intValue(), parameters.get(1).getLongLiteral().intValue());
         }
 
-        return createParametrizedDecimal(parameters);
-    }
-
-    @NotNull
-    private Type createParametrizedDecimal(List<TypeParameter> parameters)
-    {
-        checkArgument(parameters.size() == 2, "Expected 2 literal parameters for DECIMAL");
-        TypeParameter precision = parameters.get(0);
-        TypeParameter scale = parameters.get(1);
-        checkArgument(precision.isLongLiteral() &&
-                scale.isLongLiteral(), "Expected both literal parameters for DECIMAL to be numbers");
-
-        return DecimalType.createDecimalType(precision.getLongLiteral().intValue(), scale.getLongLiteral().intValue());
+        return result;
```

Comment:
Then I would have to handle null sanity check manually instead of relaying on @NotNull annotation, that's why I've changed it to not to make use of return early strategy.